src/interfaces/IPoolInitializer.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+
+interface IPoolInitializer {
+    /// @notice Initialize a Uniswap v4 Pool
+    /// @dev If the pool is already initialized, this function will not revert and just return type(int24).max
+    /// @param key the PoolKey of the pool to initialize
+    /// @param sqrtPriceX96 the initial sqrtPriceX96 of the pool
```

Comment:
nit: It can be capitalized for better readability
```suggestion
    /// @param key The PoolKey of the pool to initialize
    /// @param sqrtPriceX96 The initial sqrtPriceX96 of the pool
```